File: addon_customization/addon_customization/report/bank_payment_upload/bank_payment_upload.js
// Copyright (c) 2016, riconova and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Bank Payment Upload"] = {
	"filters": [
		{
			fieldname:"bank_format",
			label: __("Bank Format"),
			fieldtype: "Select",
			options: "Zenith Bank Format\nFidelity Bank Format\nAccess Bank Format",
			reqd : 1
			// default: frappe.datetime.add_days(frappe.datetime.month_start(),-1),
			// fetch_from: frappe.query_report.get_filter('project').value.sales_order
		},
		{
			fieldname:"bank_account",
			label: __("Bank Account"),
			fieldtype: "Link",
			options: "Account",
			reqd : 1,
			"get_query": function () {
				return {
					filters: [
						["Account", "account_type", "=", "Bank"],
						["Account", "is_group", "=", "0"]
					]
				}
			},
			on_change: function(query_report) 
			{
				if(query_report.get_values().bank_account){
					var bank_account = query_report.get_values().bank_account;
					// frappe.msgprint(project);
					frappe.model.get_value('Account', {'name': bank_account}, ['bank_account_no'],
						function(d) {
							var bank_account_no = d['bank_account_no']
							frappe.query_report.set_filter_value("company_account", bank_account_no)
							
						}
					)
				}
					
			}
			// default: frappe.datetime.add_days(frappe.datetime.month_start(),-1),
			// fetch_from: frappe.query_report.get_filter('project').value.sales_order
		},
		{
			fieldname:"payroll_entry",
			label: __("Payroll Entry"),
			fieldtype: "Link",
			options: "Payroll Entry",
			reqd : 1,
			"get_query": function () {
				return {
					filters: [
						["Payroll Entry", "docstatus", "=", "1"]
					]
				}
			},
			// default: frappe.datetime.add_days(frappe.datetime.month_start(),-1),
			// fetch_from: frappe.query_report.get_filter('project').value.sales_order
		},
		{
			fieldname:"today_date",
			label: __("Today Date"),
			fieldtype: "Date",
			default: frappe.datetime.nowdate()
			// default: frappe.datetime.add_days(frappe.datetime.month_start(),-1),
			// fetch_from: frappe.query_report.get_filter('project').value.sales_order
		},
		{
			fieldname:"company_account",
			label: __("Company Account"),
			fieldtype: "Data",
			read_only: 1
			// default: frappe.datetime.add_days(frappe.datetime.month_start(),-1),
			// fetch_from: frappe.query_report.get_filter('project').value.sales_order
		},

	]
};

File: addon_customization/addon_customization/report/bank_payment_upload/bank_payment_upload.py
# Copyright (c) 2013, riconova and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe

def execute(filters=None):
	columns, data, temp_data = [], [], []

	columns = [
		"TRANSACTION REFERENCE NUMBER:Data:200",
		"BENEFICIARY NAME:Data:200",
		"AMOUNT:Currency:150",
		"PAYMENT DUE DATE:Data:150",
		"BENEFICIARY CODE:Data:150",
		"BENEFICIARY ACCT NUMBER:Data:200",
		"BRANCH SORT CODE(9 DIGITS):Data:200",
		"DEBIT ACCT:Data:100",
		]


	payroll_entry = ""
	if filters.get("payroll_entry") :
		payroll_entry = """ AND pe.`name` = "{}" """.format(filters.get("payroll_entry"))


	bank_account = ""
	if filters.get("bank_account") :
		bank_account = """ AND pe.`payment_account` = "{}" """.format(filters.get("bank_account"))


	temp_data = frappe.db.sql("""
		SELECT

		sp.`name`,
		sp.`employee_name`,
		sp.`net_pay`,
		CURDATE(),
		sp.`employee`,
		e.`bank_ac_no`,
		pe.`payment_account`,
		e.`sort_code`

		FROM `tabPayroll Entry` pe
		LEFT JOIN `tabSalary Slip` sp ON pe.`name` = sp.`payroll_entry`
		LEFT JOIN `tabEmployee` e ON sp.`employee` = e.`name`

		WHERE pe.`docstatus` = 1
		AND sp.`docstatus` = 1

		{}
		{}

		ORDER BY pe.`name`, sp.`employee`

	""".format(payroll_entry, bank_account))

	if temp_data :
		for i in temp_data :
			sort_code = ""
			bank_account_no = ""
			if i[6] :
				bank_account_no = frappe.get_doc("Account", str(i[6])).bank_account_no


			if i[7] :
				data.append([ str(i[0]).replace("Sal ",""), i[1], i[2], str(i[3]).split("-")[2]+"/"+str(i[3]).split("-")[1]+"/"+str(i[3]).split("-")[0], str(i[4]), str(i[5]), str(i[7]), str(bank_account_no) ])
			else :
				data.append([ str(i[0]).replace("Sal ",""), i[1], i[2], str(i[3]).split("-")[2]+"/"+str(i[3]).split("-")[1]+"/"+str(i[3]).split("-")[0], str(i[4]), str(i[5]),"", str(bank_account_no) ])






	return columns, data
